src/api/controllers/lines.js

```diff
@@ -128,17 +128,85 @@ function getStatsByEmployerId(req, res) {
+        var size = getSizeByEmployerId(req.query.employer_id);
+        var currentPlace = getMyPlaceByEmployerId(req.user._id, req.query.employer_id);
+
+        Promise.all([size, currentPlace]).then(function(values) {
+            console.log('size = ' + values[0] + ', myPlace = ' + values[1]);
             res.status(200).json({
-                "size": count
+                "size": values[0],
+                "myPlace": values[1]
             });
-        });
+        })
     }
 }
 
+//helper: given employer_id, returns the # of users in line for that employer, as a promise.
+function getSizeByEmployerId(employer_id) {
+    return Line.count({ employer_id: employer_id }).exec();
+}
+
+//helper: given user_id and employer_id, returns what place this user is in line, as a promise.
+function getMyPlaceByEmployerId(user_id, employer_id) {
+    //to get current place:
+    //check current status
+    //then count # of lines with further advanced status
+    //then also add # of lines with same status, but with older update timestamps.
+    var myCurrentLine = Line.findOne({ user_id: user_id, 
+                                       employer_id: employer_id }).exec();
+    return myCurrentLine.then(function (myLine) {
+        if (!myLine) {
+            return null;
+        }
+
+        var status = myLine.status;
+
+        switch (status) {
+            case 'startrecruiter':
+                return 0; //if you're already talking to the recruiter, you're at the front.
+                break;
+
+            case 'inline':
+                var aheadOfMe = Line.count({ employer_id: employer_id,
+                                    status: 'inline',
+                                    updated_by: {$lt: myLine.updated_by}}).exec();
+                return aheadOfMe.then(function (ahead) {
+                    return ahead + 1;
+                });
+                break;
+
+            case 'notification':
+                //use exec() and Promise.all() to have queries run async. in parallel.
+                var inlineCount = Line.count({ employer_id: employer_id,
+                                               status: 'inline'}).exec(); 
+                var notifCount = Line.count({ employer_id: employer_id,
+                                              status: 'notification',
+                                              updated_by: {$lt: myLine.updated_by}}).exec();
+                return Promise.all([inlineCount, notifCount]).then(function(values) {
+                    return values.reduce(function(a, b) { return a + b; } ) + 1;
+                });
+                break;
+
+            case 'preline':
+                const aheadStatuses = ['inline', 'notification'];
+                var aheadCount = Line.count({ employer_id: employer_id, 
+                                              status: aheadStatuses}).exec();
+                var prelineCount = Line.count({ employer_id: employer_id,
```

Comment:
nice, i didn't know u could do this. we could potentially do this for all of our stuff.

Reply:
yeah if u do exec() then the query object becomes a promise